README.md
# Ubuntu Software Installer

Written By: Dale Noe (dalenoe@gmail.com)

This script was written for Ubuntu, but should work fine on Debian as well. It could also be
modified for other Linux flavors.

This script will update the software repository, before installing the packages of your choice.

## Features

It will install different types of software. Right now there is 6 things it can install.

htop  
nmap  
bmon  
lamp-server - LAMP Stack  
postgresql  
whois  
nagios3  

Want to see something added? Open a issue on the gitlab project page. Some of this
software might already be installed on your system by default. The next version will
make sure it's not installing stuff that is there by default.


CHANGELOG.md
# Version 0.1.0

This version includes a changelog, and license. This closes out issues #5 and #6.
All set for merge request !2

No code was changed.

# Version 0.0.5

The first, and quite possibly the only release.


ubuntu-webmin-install.sh
#!/bin/bash
#
# Ubuntu Webmin Installer
#
# This script is intented for a Ubuntu server, though it could be modified fairly easy.
#
# (C) Copyright 2015 Dale Noe <dalenoe@gmail.com>
#
# Changelog can be found in CHANGELOG.md
# GNU Licnese can be found in LICENSE.md
#

PASSWORD=$(whiptail --title "Root Password" --passwordbox "Enter your password and choose Ok to continue. We need the ROOT password for sudo commands." 10 60 3>&1 1>&2 2>&3)

exitstatus=$?
if [ $exitstatus = 0 ]; then

	sleep 2s

	whiptail --title "Webmin Install" --msgbox "We are about to install webmin. In the process a file (/etc/apt/sources.list.d/webmin.list) will be created for the webmin repository." 8 78

        echo "Ok, we are going to create the webmin.list file along with downloading the key"

	echo $PASSWORD | sudo bash

	sudo touch /etc/apt/sources.list.d/webmin.list |  echo -e "deb http://download.webmin.com/download/repository sarge contrib\ndeb http://webmin.mirror.somersettechsolutions.co.uk/repository sarge contrib" > /etc/apt/sources.list.d/webmin.list

	cd /root | wget http://www.webmin.com/jcameron-key.asc | sudo -S apt-key add jcameron-key.asc

	echo "Now going to update the OS package manager"

	echo "This is going to take awhile, go get a cup of coffee."

	sleep 1s

	echo $PASSWORD | sudo -S apt-get update

	sleep 1s

	echo "Phew. Ok, we made it!"

	sleep 2s

	echo "Ok.. now our packages are up to date"

	sleep 2s

	echo $PASSWORD | sudo -S apt-get install webmin

	sudo -S rm -rf jcameron-key.asc

	echo "Congrats, you have successfully install webmin"

else

	echo "You canceled. Try Again?"

fi


generic-ubuntu-install.sh
#!/bin/bash
#
# This script was written by Dale Noe (dalenoe@gmail.com)
# It's a script that installs various software
# and even makes a few changes.

password=$(whiptail --title "Root Password" --passwordbox "Enter your password and choose Ok to continue. We need the ROOT password for sudo commands." 10 60 3>&1 1>&2 2>&3)
 
exitstatus=$?
if [ $exitstatus = 0 ]; then

	sleep 2s

	echo "Now going to update the OS package manager"

	echo "This is going to take awhile, go get a cup of coffee."

	sleep 1s

#	echo $password | sudo -S apt-get update

	sleep 1s

	echo "Phew, we made it!"

	sleep 2s

	echo "Ok.. now our packages are up to date"

	echo "More shit is coming soon! lol"

		software=$(whiptail --title "Choose additional software to install." --checklist --separate-output  \
		"Choose additional software!" 15 60 4 \
		"htop" "Nice process manager" OFF \
		"nmap" "It's nmap, no explanation needed." OFF \
		"Apache2" "The apache webserver." OFF \
                "Apache4" "The apache webserver." OFF \
                "Apache5" "The apache webserver." OFF \
                "Apache6" "The apache webserver." OFF \
		"MySQL" "What doesn't need a database? lol" OFF 3>&1 1>&2 2>&3)
 
		exitstatus=$?
		if [ $exitstatus = 0 ]; then
		    echo "You chose to install: $software"
		    sudo -S apt-get install $software
#		    echo "I need to implament the commands to install the software yet."
		else
		    echo "You canceled. Not installing any software today? lol.. why did you even run me then? :P"
		fi

else

	echo "You canceled. Try Again?"

fi




generic-software-install.sh
#!/bin/bash
#
# Generic Software Installer
#
# This script is intented for a Ubuntu server, though it could be modified fairly easy.
#
# (C) Copyright 2015 Dale Noe <dalenoe@gmail.com>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
#


PASSWORD=$(whiptail --title "Root Password" --passwordbox "Enter your password and choose Ok to continue. We need the ROOT password for sudo commands." 10 60 3>&1 1>&2 2>&3)
 
exitstatus=$?
if [ $exitstatus = 0 ]; then

	sleep 2s

	echo "Now going to update the OS package manager"

	echo "This is going to take awhile, go get a cup of coffee."

	sleep 1s

#	echo $PASSWORD | sudo -S apt-get update

	sleep 1s

	echo "Phew, we made it!"

	sleep 2s

	echo "Ok.. now our packages are up to date"

	echo "More shit is coming soon! lol"

		SOFTWARE=$(whiptail --fb --title "Choose additional software to install." --checklist --separate-output  \
		"Choose additional software!" 30 120 16 \
		"htop" "Nice process manager" OFF \
		"nmap" "It's nmap, no explanation needed." OFF \
		"Apache2" "The apache webserver." OFF \
                "Apache4" "The apache webserver." OFF \
                "Apache5" "The apache webserver." OFF \
                "Apache6" "The apache webserver." OFF \
		"MySQL" "What doesn't need a database? lol" OFF 3>&1 1>&2 2>&3)
 
		exitstatus=$?
		if [ $exitstatus = 0 ]; then
		    echo "You chose to install: $SOFTWARE"
		    sudo -S apt-get install $SOFTWARE
#		    echo "I need to implament the commands to install the software yet."
		else
		    echo "You canceled. Not installing any software today? lol.. why did you even run me then? :P"
		fi

else

	echo "You canceled. Try Again?"

fi


ubuntu-software-install.sh
#!/bin/bash
#
# Ubuntu Software Installer
#
# This script is intented for a Ubuntu server, though it could be modified fairly easy.
#
# (C) Copyright 2015 Dale Noe <dalenoe@gmail.com>
#
# Changelog can be found in CHANGELOG.md
# GNU Licnese can be found in LICENSE.md
#

PASSWORD=$(whiptail --title "Root Password" --passwordbox "Enter your password and choose Ok to continue. We need the ROOT password for sudo commands." 10 60 3>&1 1>&2 2>&3)

exitstatus=$?
if [ $exitstatus = 0 ]; then

	sleep 2s

	echo "Now going to update the OS package manager"

	echo "This is going to take awhile, go get a cup of coffee."

	sleep 1s

	echo $PASSWORD | sudo -S apt-get update

	sleep 1s

	echo "Phew, we made it!"

	sleep 2s

	echo "Ok.. now our packages are up to date"

	sleep 2s

		SOFTWARE=$(whiptail --fb --title "Choose additional software to install." --checklist --separate-output  \
		"Choose additional software!" 30 120 16 \
		"htop" "Nice process manager" OFF \
		"nmap" "It's nmap, no explanation needed." OFF \
		"bmon" "Bandwidth Monitor" OFF \
                "lamp-server^" "LAMP Stack (apache, mysql, php)" OFF \
                "postgresql" "PostgreSQL" OFF \
                "whois" "A domain whois server/search." OFF \
		"nagios3" "Network/Server Monitor" OFF 3>&1 1>&2 2>&3)

		exitstatus=$?
		if [ $exitstatus = 0 ]; then
		    echo "You chose to install: $SOFTWARE"
		    sudo -S apt-get install $SOFTWARE
		else
		    echo "You canceled. Not installing any software today?"
		fi

else

	echo "You canceled. Try Again?"

fi
